docs/onboarding/ck.hiring_process.how_to_guide.md

```diff
@@ -0,0 +1,216 @@
+# Hiring process
+
+<!-- toc -->
+
+- [Hiring stages](#hiring-stages)
+- [HiringMeister](#hiringmeister)
+- [Step by step](#step-by-step)
+- [Warm-up tasks](#warm-up-tasks)
+- [Take-home message for interns](#take-home-message-for-interns)
+- [Intern scoring](#intern-scoring)
+
+<!-- tocstop -->
+
+## Hiring stages
+
+- We separate the hiring process into the following stages:
+- **Intern**
+  - Not a part of the company
+  - Paid, but not through the company payroll
+  - Interns only have access to public repos (e.g., `helpers`, `tutorials`)
+  - Can be full-time or part-time
+    - Since most are students, during summer they can work full-time
+    - In any case we expect at least 20 hrs/week
+  - Interns go through
+    ["light" on-boarding](/docs/onboarding/intern.onboarding_checklist.reference.md)
+    - Read the "must-read" docs about how we do things
+    - Set up for development on their laptops
+  - Interns get [simple tasks](#warm-up-tasks) to work on, so that we can assess
+    how well they can follow directions and work independently
+    - The tasks would be most likely related to Linter, documentation, AI, DS
+    - How to test potential hires to the Infra team? We decide case by case
+  - Ideal goal: minimal effort for us to give a chance to somebody, high
+    probability of failing
+  - Interns are managed only by the hiring team without any input from team
+    leaders
+  - After 1 month, we make a decision whether the intern can move on to the
```

Comment:
- Interns receive feedback on various skills every 2 weeks

FYI there are some docs on the procedure, e.g., https://docs.google.com/spreadsheets/d/1FCuM3dPP3QZCT7iOHSKXls-maSZ0Mwj7BjsbaUq-T0Y/edit?gid=1816709202#gid=1816709202